### PR TITLE
convert issue template to form

### PR DESCRIPTION
**Pull request by michelleN**
- Was inspired by the ORAS issue form and specifically that they have a checkbox for indicating whether the person filing the issue is interested in contributing. Would love to know what folks think about this? https://github.com/oras-project/oras/issues/new?assignees=&labels=enhancement%2Ctriage&projects=&template=feature-request.yaml

- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/suggestion.md [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
-
+body:
+  - type: markdown
+    id: preface
+    attributes:
+      value: "Thank you for submitting a suggestion for Spin!"
+  - type: input
+    id: version
+    attributes:
+      label: "What is the version of your Spin CLI?"
```
Comment (rylev) [inline .github/ISSUE_TEMPLATE/suggestion.md:13]: This is the only question that I'm unsure about. Why is this necessary to know? Are we trying to ensure that feature requests are coming from users of the latest version of Spin?

**Review by rylev — APPROVED** [review]
Looks like a nice thing to try out! I just had one uncertainity about one of the questions.

File: .github/ISSUE_TEMPLATE/suggestion.md (hunk 1 above)
Comment (michelleN) [reply .github/ISSUE_TEMPLATE/suggestion.md:13]: Yeah, exactly.
